.github/workflows/go.yaml

```diff
@@ -0,0 +1,59 @@
+name: Go
+
+on:
+  push:
+    branches: ['*']
+    tags: ['v*']
+  pull_request:
+    branches: ['*']
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        go: ["1.14.x", "1.15.x"]
+        include:
+        - go: 1.15.x
+          latest: true
+
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Load cached dependencies
+        uses: actions/cache@v1
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Download dependencies
+        run: go mod download
+
+      - name: Lint
+        if: matrix.latest
+        run: make lint
+
+      - name: Test
+        run: make test
+
+      # TODO decide whether we want to benchmark on every run
```

Comment:
This is somewhat slow currently - will probably have to skip, and manage manually.